HelloWorldFunction/src/main/java/helloworld/App.java

```diff
@@ -0,0 +1,63 @@
+package helloworld;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.net.URL;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ResourceNotFoundException;
+import java.util.ArrayList;
+/**
+ * Handler for requests to Lambda function.
+ */
+public class App implements RequestHandler<Object, Object> {
+
+    private final String TABLE_NAME = System.getenv("TABLE_NAME");
+    public Object handleRequest(final Object input, final Context context) {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", "application/json");
+        headers.put("X-Custom-Header", "application/json");
+        try {
+            final String pageContents = this.getPageContents("https://checkip.amazonaws.com");
+            this.storeLocation(pageContents);
+            String output = String.format("{ \"message\": \"hello world\", \"location\": \"%s\" }", pageContents);
+            return new GatewayResponse(output, headers, 200);
+        } catch (Exception e) {
+            return new GatewayResponse("{}", headers, 500);
+        }
+    }
+
+    private String getPageContents(String address) throws IOException {
+        URL url = new URL(address);
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()))) {
+            return br.lines().collect(Collectors.joining(System.lineSeparator()));
+        }
+    }
+
+    private void storeLocation(String ipv4) throws AmazonServiceException {
+        String now = ZonedDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
+
+        HashMap<String,AttributeValue> item_values =
+            new HashMap<String,AttributeValue>();
+
+        item_values.put("location", new AttributeValue(ipv4));
+        item_values.put("date", new AttributeValue(now));
+
+        final AmazonDynamoDB ddb = AmazonDynamoDBClientBuilder.defaultClient();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If you don’t explicitly set an AWS region, the SDK consults the default region provider chain to try and determine the region to use.  However, the resulting delay might lead to a poor cold-start. If code portability is not an issue, consider explicitly setting the AWS region using the *withRegion()* method.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html#automatically-determine-the-aws-region-from-the-environment)